plugin/completion.py

```diff
@@ -243,6 +243,24 @@ def format_completion(self, item: dict) -> 'Tuple[str, str]':
                 hint = completion_item_kind_names[kind]
         # label is an alternative for insertText if insertText not provided
         insert_text = item.get("insertText") or label
+        # try to handle textEdit if present
+        text_edit = item.get("textEdit")
+        if text_edit:
+            edit_range, edit_text = text_edit.get("range"), text_edit.get("newText")
+            if edit_range and edit_text:
+                range_start, range_end = edit_range.get("start"), edit_range.get("end")
+                if range_start and range_end:
+                    start_row, start_col = range_start.get("line"), range_start.get("character")
+                    end_row, end_col = range_end.get("line"), range_end.get("character")
+                    if all([start_row is not None, start_col is not None,
+                            end_row is not None, end_col is not None]):
```

Comment:
The range is probably specified to be required, in which case I like to avoid the None checks (I'm fine with errors from non-compliant servers).
There is existing logic for parsing ranges in `protocol.py`